mismi-core/src/Mismi/Environment.hs

```diff
@@ -40,5 +49,13 @@ getRegionFromEnv = do
     (either (throwM . RegionUnknown . T.pack) return . fromText . T.pack)
     mr
 
+discoverAwsEnv :: IO (Either EnvError Env)
+discoverAwsEnv = runEitherT $ do
+  r <- EitherT . fmap (maybeToRight MissingRegion) $ getRegionFromEnv
+  lift $ newEnv r Discover
```

Comment:
Sorry you can ignore the above comment. I see amazonka now supports this with `AWS_SESSION_TOKEN`